arch/risc-v/src/esp32c3/esp32c3.h

```diff
@@ -0,0 +1,49 @@
+/****************************************************************************
+ * arch/risc-v/src/esp32c3/esp32c3.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_RISCV_SRC_ESP32C3_ESP32C3_H
+#define __ARCH_RISCV_SRC_ESP32C3_ESP32C3_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <arch/irq.h>
+#include "riscv_internal.h"
+#include "riscv_arch.h"
+#include "chip.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef __ASSEMBLY__
+
+#define setbits(bs, a)     modifyreg32(a, 0, bs)
+#define resetbits(bs, a)   modifyreg32(a, bs, 0)
```

Comment:
I'm not sure if it is the best to hide modifyreg32 behind a macro, as this would deviate from other arch/SoC support code. I'm not fully against it but maybe a better approach would be to provide this for all arch instead.

Reply:
Actually, setbits is basically putreg32.

Reply:
> 
> 
> Actually, setbits is basically putreg32.

Not really modifyreg32(reg, 0, bs) would expand to:

    putreg32(reg, (getreg32(reg) & ~0) | bs));

or just

    putreg32(reg, getreg32(req) | bs);

The bits in bs are set.  Other bits that are set are retained.  No bits are cleared.

    putreg32(reg, bs)

on the other hand, will set all bits in bs and clear all bits not in bs

Reply:
Right. So maybe we can provide these for convenience later to all archs as setbits32/resetbits32 (or clearbits32).